amsclient/utils.go

```diff
@@ -17,21 +17,61 @@ package amsclient
 import (
 	"fmt"
 	"strings"
+
+	"github.com/RedHatInsights/insights-results-smart-proxy/types"
+)
+
+const (
+	searchQuerySeparator = "','"
 )
 
 // generateSearchParameter generates a search string for given org_id and desired statuses
 func generateSearchParameter(orgID string, allowedStatuses, disallowedStatuses []string) string {
 	searchQuery := fmt.Sprintf("organization_id is '%s' and cluster_id != ''", orgID)
 
 	if len(allowedStatuses) > 0 {
-		clusterIDQuery := " and status in ('" + strings.Join(allowedStatuses, "','") + "')"
+		clusterIDQuery := " and status in ('" + strings.Join(allowedStatuses, searchQuerySeparator) + "')"
+		searchQuery += clusterIDQuery
+	}
+
+	if len(disallowedStatuses) > 0 {
+		clusterIDQuery := " and status not in ('" + strings.Join(disallowedStatuses, searchQuerySeparator) + "')"
+		searchQuery += clusterIDQuery
+	}
+
+	return searchQuery
+}
+
+// generateMulticlusterSearchQuery generates a search string for given org_id, list of clusters and desired statuses
+func generateMulticlusterSearchQuery(orgID string, clusterIDs []string, allowedStatuses, disallowedStatuses []string) string {
+	searchQuery := fmt.Sprintf("organization_id is '%s'", orgID)
+
+	if len(clusterIDs) > 0 {
+		clusterIDQuery := " and cluster_id in ('" + strings.Join(clusterIDs, searchQuerySeparator) + "')"
+		searchQuery += clusterIDQuery
+	}
+
+	if len(allowedStatuses) > 0 {
```

Comment:
isn't this a duplication of what is defined in `generateSearchParameter`?

Reply:
yep, I can refactor the allowed/disallowed part. 

The first part is different:
- generateSearchParameter simply ignores results without cluster IDs (cluster_id != '')
- generateMulticlusterSearchQuery uses a `cluster_id IN` clause.

Reply:
removed all these, as there's no need to query ACM in the end